tracer/src/Datadog.Trace/Propagators/W3CTraceContextPropagator.cs

```diff
@@ -442,7 +461,12 @@ internal static W3CTraceState ParseTraceState(string header)
                     propagatedTags = null;
                 }
 
-                return new W3CTraceState(samplingPriority, origin, propagatedTags, additionalValues);
+                if (lastParent is null)
+                {
+                    lastParent = "0000000000000000";
```

Comment:
So if the `lastParent` is `null` I'm essentially treating that later on as the local context being created by `datadog` headers.
Which if it is created by `datadog` headers then the `_dd.parent_id` tag shouldn't be set, but this will change in the next PR (very soon) where if the traceIDs of a `datadog` and `tracecontext` header match then the `lastParent` and `_dd.parent_id` will always be added.

ID of `0` here also means that the backend could reparent this span as the root in the trace if needed.